rvps/src/cache/local_fs/mod.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! This Cache stores RV information inside a local file
+
+use std::path::Path;
+
+use anyhow::*;
+
+use crate::{Cache, ReferenceValue};
+
+/// Local directory path to store the reference values,
+/// which is created by sled engine.
+const FILE_PATH: &str = "./reference_values";
+
+/// `LocalFs` implements [`Cache`] trait. And
+/// it uses rocksdb inside.
+pub struct LocalFs {
+    engine: sled::Db,
+}
+
+impl Default for LocalFs {
+    fn default() -> Self {
+        LocalFs::new(FILE_PATH).expect("Failed to create LocalFs Cache.")
+    }
+}
+
+impl LocalFs {
+    /// Create a new [`LocalFs`] with given
+    /// file storage path.
+    fn new(path: &str) -> Result<Self> {
+        let path = Path::new(path);
+        let engine = sled::open(path)?;
+        Ok(Self { engine })
+    }
+}
+
+impl Cache for LocalFs {
+    fn set(&mut self, name: String, rv: ReferenceValue) -> Result<Option<ReferenceValue>> {
+        let rv_serde = serde_json::to_string(&rv)?;
+        println!("{}", rv_serde);
+        match self.engine.insert(name, rv_serde.as_bytes())? {
+            Some(v) => {
+                self.engine.flush()?;
+                let v = serde_json::from_slice(&v)?;
+                Ok(Some(v))
+            }
+            None => Ok(None),
+        }
```

Comment:
Here we flush if there a previous value already existed in the DB, but we don't flush if it's a new key-value pair. That doesn't seem like the correct logic.

Also, is there a use to returning the old value? Is that the expected behaviour of `Cache::set()` ?
If there's no use, maybe we can return `Result<()>` (here and in `cache/mod.rs`).

Reply:
Yes, it is the expected behavior of `Cache::set()`. I've add some more notes on the trait.

Reply:
> > Also, is there a use to returning the old value? Is that the expected behaviour of `Cache::set()` ?
> > If there's no use, maybe we can return `Result<()>` (here and in `cache/mod.rs`).
> 
> Yes, it is the expected behavior of `Cache::set()`. I've add some more notes on the trait.

If that's the expected behaviour, it should be tested in the unit-tests. It seems that in `set_and_get` we ignore the return value of `.set()`. Consider adding a test for setting a new key and for setting an existing key, and checking the return value in both cases.

Reply:
Yes, you're right. I've added it, thanks!